src/backend/distributed/planner/query_colocation_checker.c

```diff
@@ -434,6 +473,25 @@ CreateTargetEntryForNullCol(Form_pg_attribute attributeTuple, int resno)
 }
 
 
+/*
+ * CreateTargetEntryForDroppedCol creates a dummy column for the dropped column.
+ */
+static TargetEntry *
+CreateTargetEntryForDroppedCol(Form_pg_attribute attributeTuple, int resno)
+{
+	Expr *nullExpr = (Expr *) makeNullConst(attributeTuple->atttypid,
+											attributeTuple->atttypmod,
+											attributeTuple->attcollation);
```

Comment:
why do we use values from a completely unrelated column? could we just set this to some arbitrary constants? (e.g. use INT4OID)

Reply:
I guess we could do this, maybe we can do the similar for null target entries, do you think that would be confusing?

Reply:
at least for dropped columns it would be less confusing than taking some random column

for unused columns, either way works I think

Reply:
I left the other part as is